### ISSUE TITLE
Spigot server in Web mode produces error during "player.message" (OP player)

### ISSUE THREAD

**Issue by lukeeexd** [benbaptist/minecraft-wrapper#504]
When I type something I get this error.

```[21:50:23 INFO]: <Luke> t
[21:50:23] [Wrapper.py/ERROR]: Plugin 'Web'
experienced an exception calling 'player.message':
'bool' object has no attribute 'username'
Traceback (most recent call last):
  File "Wrapper.py/core/events.py", line 65, in callevent
    result = self.events[plugin_id][event](payload)
  File "Wrapper.py/management/web.py", line 84, in onPlayerMessage
    "player": payload["player"].username,
AttributeError: 'bool' object has no attribute 'username'
```

**Comment by suresttexas00** [comment]
Web mode is broken presently; although I will look into the specific error, web mode should be turned off in wrapper.properties.json.

**Comment by lukeeexd** [comment]
Was pretty certain I had web mode off. For some reason it was turned on.

**Comment by suresttexas00** [comment]
I actually cannot see a problem with that part of the code.  It appears as though the player object is not getting created at login.  If you would, I would like for you to:
1) re-enable web mode.
2) set your console output to "DEBUG". (logging.json, under["handlers"]["console"])
3) re-start wrapper.
4) Log in and reproduce the error again.
5) Give me a cut and paste of the _entire_ console output from start to finish. enclosed in \`\`\`

I cannot reproduce the error, making me think this is a spigot parsing issue again.

**Comment by suresttexas00** [comment]
forget all that.. There is an error after all.  player.message passes the player name only, not the playerobject.  Therefore, Web module is using the API improperly.  This is weird because the IRC uses it correctly.  Both code bases are still primarily Ben's.

**Comment by suresttexas00** [comment]
I am still interested in this error because it seems to be a symptom of a possibly larger problem..  I believe the player object is not getting logged in.  If you can re-enable web mode again, log on, chat, and then give me a paste of the whole console output (not the logs) from start to finish, I'd appreciate it!  (I was not able to replicate this error and believe it is Spigot-related)

**Comment by lukeeexd** [comment]
the error only shows when someone who is OP talks

**Comment by suresttexas00** [comment]
I was not able to replicate it even though I am an OP on my servers, so that console readout will be helpful.

**Comment by lukeeexd** [comment]
```[01:16:48 INFO]: UUID of player lukeeexd is 03a588b1-6163-4dd7-b26d-f249ed2bbe5b
[01:16:48 INFO]: lukeeexd[/] logged in with entity id 142 at ([world]316.86789318152546, 67.12426603789697, -191.9069627257038)
[01:16:58 INFO]: <Luke> t
[01:16:58] [Wrapper.py/ERROR]: Plugin 'Web'
experienced an exception calling 'player.message':
'bool' object has no attribute 'username'
Traceback (most recent call last):
  File "Wrapper.py/core/events.py", line 65, in callevent
    result = self.events[plugin_id][event](payload)
  File "Wrapper.py/management/web.py", line 84, in onPlayerMessage
    "player": payload["player"].username,
AttributeError: 'bool' object has no attribute 'username'
```

**Comment by suresttexas00** [comment]
I need more.. I need the whole console output from the time wrapper starts until you get the error, for example (most consoles you can SHIFT-CTR-C to copy a console selection:

example:
```
[23:23:53] [root/INFO]: Logging configuration file (logging.json) located and loaded, logging configuration set!
[23:23:53] [Wrapper.py/INFO]: Wrapper.py started - Version [0, 15, 0] (development build #255)
[23:23:53] [Wrapper.py/DEBUG]: Wrapper is using Python 2.7.
[23:23:53] [Wrapper.py/DEBUG]: [Wrapper.py] Registered help group 'Wrapper' with 14 commands
[23:23:53] [Wrapper.py/INFO]: Loading plugins...
[23:23:53] [Wrapper.py/DEBUG]: NAME OF PLUGIN: SurestLib
[23:23:53] [Wrapper.py/DEBUG]: Reading plugin file SurestLib.py ...
[23:23:53] [Wrapper.py/DEBUG]: Plugin 'SurestLib' disabled - not loading
[23:23:53] [Wrapper.py/DEBUG]: NAME OF PLUGIN: homes
[23:23:53] [Wrapper.py/DEBUG]: Reading plugin file homes.py ...
[23:23:53] [Wrapper.py/DEBUG]: Loading plugin homes...
[23:23:53] [Wrapper.py/DEBUG]: [homes] Registered help group 'homes' with 6 commands
[23:23:53] [Wrapper.py/DEBUG]: [homes] Registered command 'sethome'
[23:23:53] [Wrapper.py/DEBUG]: [homes] Registered command 'home'
[23:23:53] [Wrapper.py/DEBUG]: [homes] Registered command 'homes'
[23:23:53] [Wrapper.py/DEBUG]: [homes] Registered event 'player.message'
[23:23:53] [Wrapper.py/INFO]: Plugin homes loaded...
[23:23:53] [Wrapper.py/INFO]: Connecting to IRC...
[23:23:53] [Wrapper.py/WARNING]: Our apologies!  Web mode is currently broken.  Wrapper will start web mode anyway, but it will not likely function well (or at all).  For now, you should turn off web mode in wrapper.properties.json.
[23:23:53] [Web/INFO]: Web Interface bound to 0.0.0.0:8070
[23:23:53] [Wrapper.py/INFO]: Starting server...
[23:23:53] [Wrapper.py/DEBUG]: Checking EULA agreement...
[23:23:53] [Wrapper.py/DEBUG]: EULA agreement has been accepted.
[23:23:56] [Wrapper.py/DEBUG]: [Wrapper.py] Registered permission 'mc1.7.6' with default value: True
[23:23:56] [Server thread/INFO]: Starting minecraft server version 1.12.2
[23:23:56] [Server thread/INFO]: Loading properties
[23:23:56] [Server thread/INFO]: Default game type: SURVIVAL
[23:23:56] [Server thread/INFO]: Generating keypair
[23:23:56] [Server thread/INFO]: Starting Minecraft server on *:25542
[23:23:56] [Server thread/INFO]: Using epoll channel type
[23:23:56] [Server thread/WARN]: **** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
[23:23:56] [Server thread/WARN]: The server will make no attempt to authenticate usernames. Beware.
[23:23:56] [Server thread/WARN]: While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.
[23:23:56] [Server thread/WARN]: To change this, set "online-mode" to "true" in the server.properties file.
[23:23:56] [Server thread/INFO]: Preparing level "world"
[23:23:56] [Server thread/INFO]: Loaded 488 advancements
[23:23:56] [Server thread/INFO]: Preparing start region for level 0
[23:23:57] [Server thread/INFO]: Preparing spawn area: 81%
[23:23:57] [Server thread/INFO]: Done (1.347s)! For help, type "help" or "?"
[23:23:57] [Wrapper.py/INFO]: Server started
[23:24:05] [Wrapper.py/INFO]: Connected to IRC!
[23:24:05] [Server thread/INFO]: Unknown command. Try /help for a list of commands
[23:24:34] [Server thread/INFO]: SurestTexas00[/127.0.0.1:47434] logged in with entity id 149 at (46.29907483845001, 63.0, -270.1293488726086)
[23:24:34] [Server thread/INFO]: SurestTexas00 joined the game
[23:24:42] [Server thread/INFO]: <SurestTexas00> !home
[23:24:42] [Server thread/INFO]: Teleported SurestTexas00 to 65.5, 69.0, -267.5
[23:24:42] [Server thread/WARN]: SurestTexas00 moved too quickly! 19.20092516154999,6.0,2.6293488726086025
[23:24:49] [Server thread/INFO]: <SurestTexas00> its dark
[23:25:19] [Server thread/INFO]: SurestTexas00 tried to swim in lava
[23:25:29] [Server thread/INFO]: <SurestTexas00> !home
[23:25:30] [Server thread/INFO]: Teleported SurestTexas00 to 65.5, 69.0, -267.5
[23:25:30] [Server thread/WARN]: SurestTexas00 moved too quickly! 183.0,3.0,-88.0
[23:25:33] [Server thread/INFO]: SurestTexas00 lost connection: Disconnected
[23:25:33] [Server thread/INFO]: SurestTexas00 left the game
```

**Comment by suresttexas00** [comment]
Also, set your wrapper console logging to "DEBUG" in logging.json:

```
        "console": {
            "stream": "ext://sys.stdout",
            "formatter": "standard",
            "class": "logging.StreamHandler",
            "filters": [],
            "level": "DEBUG"
        },
```

**Comment by suresttexas00** [comment]
Looks like you may have given me a clue above, but I will need to verify it with a full readout...

**Comment by suresttexas00** [comment]
I noticed this`[/]` next to your name in the login: `INFO]: lukeeexd[/] log...`

**Comment by lukeeexd** [comment]
Ok I’ll get the whole console. The [/] is where my IP is, I just removed it

**Comment by lukeeexd** [comment]
```[12:13:00] [root/INFO]: Logging configuration file (logging.json) located and loaded, logging configuration set!
[12:13:01] [Wrapper.py/INFO]: Wrapper.py started - Version [0, 14, 2] (development build #251)
[12:13:01] [Wrapper.py/INFO]: Loading plugins...
[12:13:01] [Web/INFO]: Web Interface bound to 0.0.0.0:8071
Exception in thread Thread-5:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
TypeError: eachsecond_web() takes exactly 2 arguments (1 given)

[12:13:01] [Wrapper.py/INFO]: Starting server...
Loading libraries, please wait...
2018-01-05 12:13:01,462 main WARN Disabling terminal, you're running in an unsupported environment.
[12:13:02 INFO]: Starting minecraft server version 1.12.2
[12:13:02 INFO]: Loading properties
[12:13:02 INFO]: Default game type: SURVIVAL
[12:13:02 INFO]: This server is running Paper version git-Paper-1284 (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)
[12:13:02 INFO]: Server Ping Player Sample Count: 12
[12:13:02 INFO]: Using 4 threads for Netty based IO
[12:13:02 INFO]: Debug logging is disabled
[12:13:02 INFO]: Generating keypair
[12:13:03 INFO]: Starting Minecraft server on 188.165.251.221:25569
[12:13:03 INFO]: Using epoll channel type
[12:13:03 INFO]: Set PluginClassLoader as parallel capable
[12:13:03 INFO]: [WorldEdit] Loading WorldEdit v6.1.8-SNAPSHOT;cd4729f
[12:13:03 INFO]: [SeePlayerInventory] Loading SeePlayerInventory v2.2
[12:13:03 INFO]: [NoPoppyDrop] Loading NoPoppyDrop v1.0
[12:13:03 INFO]: [CustomWhitelist] Loading CustomWhitelist v1.4
[12:13:03 INFO]: [NameColor] Loading NameColor v1.3
[12:13:03 INFO]: [Nicky] Loading Nicky v0.6.9
[12:13:03 INFO]: [PlaceholderAPI] Loading PlaceholderAPI v2.8.2
[12:13:03 INFO]: [TWSConfiguration] Loading TWSConfiguration v3.0.1
[12:13:03 INFO]: [PacketListenerApi] Loading PacketListenerApi v3.6.0-SNAPSHOT
[12:13:03 INFO]: [ReflectionHelper] Version is v1_12_R1 (11201)
[12:13:03 INFO]: [PacketListenerAPI] Using INChannel
[12:13:03 INFO]: [PacketListenerAPI] Injected custom channel handlers.
[12:13:03 INFO]: [ClearLag] Loading ClearLag v2.9.7
[12:13:03 INFO]: [DragonDeathDrops] Loading DragonDeathDrops v1.0.0
[12:13:03 INFO]: [PermissionsEx] Loading PermissionsEx v1.23.4
[12:13:03 INFO]: [KeepChunks] Loading KeepChunks v1.3.8
[12:13:03 INFO]: [InactivityChecker] Loading InactivityChecker v1.0.0
[12:13:03 INFO]: [RelaAutoBroadcast] Loading RelaAutoBroadcast v1.9
[12:13:03 INFO]: [Tablist] Loading Tablist v1.3.2.9
[12:13:03 INFO]: [StaffPlus] Loading StaffPlus v3.2.3
[12:13:03 INFO]: [CPCameraStudioReborn] Loading CPCameraStudioReborn v1.4.2
[12:13:03 INFO]: [ProtocolLib] Loading ProtocolLib v4.3.0
[12:13:03 WARN]: [ProtocolLib] Version (MC: 1.12.2) has not yet been tested! Proceed with caution.
[12:13:03 INFO]: [MCBans] Loading MCBans v4.4.3
[12:13:03 INFO]: Server permissions file permissions.yml is empty, ignoring it
[12:13:03 INFO]: [ProtocolLib] Enabling ProtocolLib v4.3.0
[12:13:03 INFO]: [ProtocolLib] Started structure compiler thread.
[12:13:03 INFO]: **** Beginning UUID conversion, this may take A LONG time ****
[12:13:03 INFO]: Preparing level "world"
[12:13:03 INFO]: -------- World Settings For [world] --------
[12:13:03 INFO]: Cactus Growth Modifier: 100%
[12:13:03 INFO]: Cane Growth Modifier: 100%
[12:13:03 INFO]: Melon Growth Modifier: 100%
[12:13:03 INFO]: Mushroom Growth Modifier: 100%
[12:13:03 INFO]: Pumpkin Growth Modifier: 100%
[12:13:03 INFO]: Sapling Growth Modifier: 100%
[12:13:03 INFO]: Wheat Growth Modifier: 100%
[12:13:03 INFO]: NetherWart Growth Modifier: 100%
[12:13:03 INFO]: Vine Growth Modifier: 100%
[12:13:03 INFO]: Cocoa Growth Modifier: 100%
[12:13:03 INFO]: Entity Activation Range: An 32 / Mo 100 / Mi 16
[12:13:03 INFO]: Entity Tracking Range: Pl 48 / An 48 / Mo 100 / Mi 32 / Other 64
[12:13:03 INFO]: Mob Spawn Range: 6
[12:13:03 INFO]: Hopper Transfer: 8 Hopper Check: 1 Hopper Amount: 1
[12:13:03 INFO]: Random Lighting Updates: false
[12:13:03 INFO]: Structure Info Saving: true
[12:13:03 INFO]: Custom Map Seeds:  Village: 10387312 Feature: 14357617 Monument: 10387313 Slime: 987234911
[12:13:03 INFO]: Max TNT Explosions: 100
[12:13:03 INFO]: Tile Max Tick Time: 50ms Entity max Tick Time: 50ms
[12:13:03 INFO]: Item Despawn Rate: 6000
[12:13:03 INFO]: Item Merge Radius: -1.0
[12:13:03 INFO]: View Distance: 13
[12:13:03 INFO]: Arrow Despawn Rate: 1200
[12:13:03 INFO]: Allow Zombie Pigmen to spawn from portal blocks: true
[12:13:03 INFO]: Experience Merge Radius: -1.0
[12:13:03 INFO]: Zombie Aggressive Towards Villager: true
[12:13:03 INFO]: Nerfing mobs spawned from spawners: false
[12:13:04 INFO]: -------- World Settings For [world_nether] --------
[12:13:04 INFO]: Cactus Growth Modifier: 100%
[12:13:04 INFO]: Cane Growth Modifier: 100%
[12:13:04 INFO]: Melon Growth Modifier: 100%
[12:13:04 INFO]: Mushroom Growth Modifier: 100%
[12:13:04 INFO]: Pumpkin Growth Modifier: 100%
[12:13:04 INFO]: Sapling Growth Modifier: 100%
[12:13:04 INFO]: Wheat Growth Modifier: 100%
[12:13:04 INFO]: NetherWart Growth Modifier: 100%
[12:13:04 INFO]: Vine Growth Modifier: 100%
[12:13:04 INFO]: Cocoa Growth Modifier: 100%
[12:13:04 INFO]: Entity Activation Range: An 32 / Mo 100 / Mi 16
[12:13:04 INFO]: Entity Tracking Range: Pl 48 / An 48 / Mo 100 / Mi 32 / Other 64
[12:13:04 INFO]: Mob Spawn Range: 6
[12:13:04 INFO]: Hopper Transfer: 8 Hopper Check: 1 Hopper Amount: 1
[12:13:04 INFO]: Random Lighting Updates: false
[12:13:04 INFO]: Structure Info Saving: true
[12:13:04 INFO]: Custom Map Seeds:  Village: 10387312 Feature: 14357617 Monument: 10387313 Slime: 987234911
[12:13:04 INFO]: Max TNT Explosions: 100
[12:13:04 INFO]: Tile Max Tick Time: 50ms Entity max Tick Time: 50ms
[12:13:04 INFO]: Item Despawn Rate: 6000
[12:13:04 INFO]: Item Merge Radius: -1.0
[12:13:04 INFO]: View Distance: 13
[12:13:04 INFO]: Arrow Despawn Rate: 1200
[12:13:04 INFO]: Allow Zombie Pigmen to spawn from portal blocks: true
[12:13:04 INFO]: Experience Merge Radius: -1.0
[12:13:04 INFO]: Zombie Aggressive Towards Villager: true
[12:13:04 INFO]: Nerfing mobs spawned from spawners: false
[12:13:04 INFO]: -------- World Settings For [world_the_end] --------
[12:13:04 INFO]: Cactus Growth Modifier: 100%
[12:13:04 INFO]: Cane Growth Modifier: 100%
[12:13:04 INFO]: Melon Growth Modifier: 100%
[12:13:04 INFO]: Mushroom Growth Modifier: 100%
[12:13:04 INFO]: Pumpkin Growth Modifier: 100%
[12:13:04 INFO]: Sapling Growth Modifier: 100%
[12:13:04 INFO]: Wheat Growth Modifier: 100%
[12:13:04 INFO]: NetherWart Growth Modifier: 100%
[12:13:04 INFO]: Vine Growth Modifier: 100%
[12:13:04 INFO]: Cocoa Growth Modifier: 100%
[12:13:04 INFO]: Entity Activation Range: An 32 / Mo 100 / Mi 16
[12:13:04 INFO]: Entity Tracking Range: Pl 48 / An 48 / Mo 100 / Mi 32 / Other 64
[12:13:04 INFO]: Mob Spawn Range: 6
[12:13:04 INFO]: Hopper Transfer: 8 Hopper Check: 1 Hopper Amount: 1
[12:13:04 INFO]: Random Lighting Updates: false
[12:13:04 INFO]: Structure Info Saving: true
[12:13:04 INFO]: Custom Map Seeds:  Village: 10387312 Feature: 14357617 Monument: 10387313 Slime: 987234911
[12:13:04 INFO]: Max TNT Explosions: 100
[12:13:04 INFO]: Tile Max Tick Time: 50ms Entity max Tick Time: 50ms
[12:13:04 INFO]: Item Despawn Rate: 6000
[12:13:04 INFO]: Item Merge Radius: -1.0
[12:13:04 INFO]: View Distance: 13
[12:13:04 INFO]: Arrow Despawn Rate: 1200
[12:13:04 INFO]: Allow Zombie Pigmen to spawn from portal blocks: true
[12:13:04 INFO]: Experience Merge Radius: -1.0
[12:13:04 INFO]: Zombie Aggressive Towards Villager: true
[12:13:04 INFO]: Nerfing mobs spawned from spawners: false
[12:13:04 INFO]: Preparing start region for level 0 (Seed: 5236083852282139121)
[12:13:04 INFO]: Preparing start region for level 1 (Seed: 1534284487993963504)
[12:13:04 INFO]: Preparing start region for level 2 (Seed: 1534284487993963504)
[12:13:05 INFO]: [WorldEdit] Enabling WorldEdit v6.1.8-SNAPSHOT;cd4729f
[12:13:05 INFO]: WEPIF: Using the Bukkit Permissions API.
[12:13:05 INFO]: [WorldEdit] Using com.sk89q.worldedit.bukkit.adapter.impl.Spigot_v1_12_R2 as the Bukkit adapter
[12:13:05 INFO]: [SeePlayerInventory] Enabling SeePlayerInventory v2.2
[12:13:05 INFO]: [SeePlayerInventory] SeePlayerInventory Enabled.
[12:13:05 INFO]: [SeePlayerInventory] Plugin by [peti446]
[12:13:05 INFO]: [SeePlayerInventory] Version: 2.2
[12:13:05 INFO]: [SeePlayerInventory] Checking for updates.............
[12:13:05 INFO]: [SeePlayerInventory] Checking for updates................
[12:13:06 WARN]: [SeePlayerInventory] There is a new version available!
[12:13:06 WARN]: [SeePlayerInventory] Your can get it from: http://servermods.cursecdn.com/files/2489/944/SeePlayerInventory-2.3.jar
[12:13:06 INFO]: [NoPoppyDrop] Enabling NoPoppyDrop v1.0
[12:13:06 INFO]: [CustomWhitelist] Enabling CustomWhitelist v1.4
[12:13:06 INFO]: [NameColor] Enabling NameColor v1.3
[12:13:06 INFO]: [Name Color] No update found!
[12:13:06 INFO]: [Nicky] Enabling Nicky v0.6.9
[12:13:06 INFO]: [Nicky] Database set to sqlite.
[12:13:06 INFO]: [PlaceholderAPI] Enabling PlaceholderAPI v2.8.2
[12:13:06 INFO]: [PlaceholderAPI] Placeholder expansion registration initializing...
[12:13:06 INFO]: [PlaceholderAPI] Successfully registered expansion: statistic
[12:13:06 INFO]: [PlaceholderAPI] Successfully registered expansion: server
[12:13:06 INFO]: [PlaceholderAPI] NMS placeholders such as %player_ping% have been enabled!
[12:13:06 INFO]: [PlaceholderAPI] Successfully registered expansion: player
[12:13:06 INFO]: [PlaceholderAPI] Fetching available expansion list from extendedclip.com...
[12:13:06 INFO]: [TWSConfiguration] Enabling TWSConfiguration v3.0.1
[12:13:06 INFO]: [PacketListenerApi] Enabling PacketListenerApi v3.6.0-SNAPSHOT
[12:13:06 INFO]: [PacketListenerAPI] Adding channels for online players...
[12:13:06 INFO]: [ClearLag] Enabling ClearLag v2.9.7
[12:13:06 INFO]: [ClearLag] Loading modules...
[12:13:06 INFO]: [ClearLag] Modules enabed, loading config values
[12:13:06 INFO]: [ClearLag] Modules have been loaded!
[12:13:06 INFO]: [ClearLag] Clearlag is now enabled!
[12:13:06 INFO]: [DragonDeathDrops] Enabling DragonDeathDrops v1.0.0
[12:13:06 INFO]: [ClearLag] Checking for updates compatible with your bukkit version [1.12]...
[12:13:06 INFO]: [PermissionsEx] Enabling PermissionsEx v1.23.4
[12:13:06 INFO]: [PermissionsEx] Initializing file backend
[12:13:06 INFO]: [PermissionsEx] Permissions file successfully reloaded
[12:13:06 INFO]: WEPIF: PermissionsEx detected! Using PermissionsEx for permissions.
[12:13:06 INFO]: [KeepChunks] Enabling KeepChunks v1.3.8


[12:13:06 INFO]: (_)   | |                 (_______) |               | |  v1.3.8

[12:13:06 INFO]: |  _   _) ___ | ___ |  _ \| |     |  _ \| | | |  _ \| |_/ )/___)
[12:13:06 INFO]: | |  \ \| ____| ____| |_| | |_____| | | | |_| | | | |  _ (|___ |
[12:13:06 INFO]: |_|   \_)_____)_____)  __/ \______)_| |_|____/|_| |_|_| \_|___/


[12:13:06 INFO]: [InactivityChecker] Enabling InactivityChecker v1.0.0
[12:13:06 INFO]: [InactivityChecker] Whitelist Inactivity Checker has successfully started!
[12:13:06 INFO]: [RelaAutoBroadcast] Enabling RelaAutoBroadcast v1.9
[12:13:06 INFO]: [Tablist] Enabling Tablist v1.3.2.9
[12:13:06 INFO]: Tablist> You have installed PlaceholderAPI feel free to use it !
[12:13:06 INFO]: Tablist> You have to install Vault to use the part of Economy
[12:13:06 INFO]: Tablist> You have to install Vault to use the part of Chat
[12:13:06 INFO]: Tablist> Tablist v1.3.2.9 enabled!!
[12:13:06 INFO]: [StaffPlus] Enabling StaffPlus v3.2.3
[12:13:06 INFO]: [Staff+] Version protocol set to 'v1_12_R1'.
[12:13:06 INFO]: [Staff+] Staff+ has been enabled! Initialization took 38ms.
[12:13:06 INFO]: [Staff+] Plugin created by Shortninja.
[12:13:06 INFO]: [CPCameraStudioReborn] Enabling CPCameraStudioReborn v1.4.2
[12:13:06 INFO]: [CPCameraStudioReborn] [CPCameraStudio] CPCameraStudioReborn has been enabled!
[12:13:06 INFO]: [MCBans] Enabling MCBans v4.4.3
[12:13:06 INFO]: [MCBans] Loading language file: default
[12:13:06 INFO]: [MCBans] Using SuperPerms for permission control.
[12:13:06 INFO]: [MCBans] Connecting to the API Server!
[12:13:06 INFO]: [MCBans] No rollback plugin not found!
[12:13:06 INFO]: [MCBans] MCBans version 4.4.3 is enabled!
[12:13:06 INFO]: Done (3.237s)! For help, type "help" or "?"
[12:13:07] [Wrapper.py/INFO]: Server started
[12:13:06 INFO]: Starting remote control listener
[12:13:06 WARN]: Unable to initialise rcon on 188.165.251.221:65535 : Address already in use (Bind failed)
[12:13:07 INFO]: Timings Reset
[12:13:07 INFO]: Join the discord server on http://www.discord.tws-mc.com | type "/discord link" and follow the instructions to link your MC account to discord!
[12:13:07 INFO]: [PacketListenerApi] Plugin is up-to-date
[12:13:07 INFO]: [ClearLag] No updates found!
[12:13:07 INFO]: [MCBans] API Server found: api.mcbans.com :: response time: 263
[12:13:07 INFO]: [MCBans] Fastest server selected: api.mcbans.com :: response time: 263
[12:13:07 INFO]: [PlaceholderAPI] 92 placeholder expansions are available on the cloud.
[12:13:07 INFO]: [PlaceholderAPI] 0 expansions you use have updates.
[12:13:18 INFO]: UUID of player lukeeexd is 03a588b1-6163-4dd7-b26d-f249ed2bbe5b
[12:13:19 INFO]: lukeeexd[/] logged in with entity id 123 at ([world]316.86789318152546, 67.12426603789697, -191.9069627257038)
[12:13:24 INFO]: <cLuke> t
[12:13:24] [Wrapper.py/ERROR]: Plugin 'Web'
experienced an exception calling 'player.message':
'bool' object has no attribute 'username'
Traceback (most recent call last):
  File "Wrapper.py/core/events.py", line 65, in callevent
    result = self.events[plugin_id][event](payload)
  File "Wrapper.py/management/web.py", line 84, in onPlayerMessage
    "player": payload["player"].username,
AttributeError: 'bool' object has no attribute 'username'
```

**Comment by suresttexas00** [comment]
Wow.  Now I see the problem and I don't know how to fix it properly.  You are logged in as lukeexd (or possibly lukeexd[/]), but the chat comes from your "nickname" cLuke.. which wrapper has no clue who that is.  Ah spigot, what a mess.

**Comment by lukeeexd** [comment]
Yeah my nickname is %cLuke to show up as Luke(in red) I used this plugin https://www.spigotmc.org/resources/nicky.590/

**Comment by suresttexas00** [comment]
The first error is fixed in 0.15.0 version (you can /update-wrapper).  I don't know how to properly deal with the second... in fact, no wrapper plugin will work if it uses the player.message event. ... which may not matter anyway since you are using spigot.  I will just have to get wrapper to ignore chat from players with no player object (i.e., the nickname).  Not an ideal way to fix it, but wrapper has no place in the console text to deduce that lukeexd = cLuke (red luke).

**Comment by suresttexas00** [comment]
#509
